msg-socket/src/req/mod.rs

```diff
@@ -27,12 +27,15 @@ pub enum ReqError {
     SocketClosed,
     #[error("Transport error: {0:?}")]
     Transport(#[from] Box<dyn std::error::Error + Send + Sync>),
+    #[error("Request timed out")]
+    Timeout,
 }
 
 pub enum Command {
     Send {
         message: Bytes,
         response: oneshot::Sender<Result<Bytes, ReqError>>,
+        expiration: Instant,
```

Comment:
This is unused and already configured in `ReqOptions`. For now, i'm fine with all requests having the same timeout.